lua/pixelui/core/cl_imgur.lua

```diff
@@ -18,17 +18,54 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 local materials = {}
 
+local imgurQueue = {}
+
+function imgurQueue.new()
+    return { coroutines = {} }
+end
+
+function imgurQueue.enqueue(queue, func, ...)
+    local co = coroutine.create(func)
+    table.insert(queue.coroutines, { co = co, args = {...} })
+end
+
+function imgurQueue.process(queue)
+    if queue.processing then return end
+    queue.processing = true
+
+    while table.Count(queue.coroutines) >= 0 do
```

Comment:
Calculate the length of `queue.coroutines` once and cache it in a variable, this is a fairly expensive operation sometimes.